False positive with MSVC and precompiled header when changing a define option

### How to reproduce ###
1. Have ccache configured to cache precompiled header (`sloppiness = time_macros`)
2. Have 3 files, PCH.h, PCH.cpp and source.cpp where both PCH.cpp and source.cpp contain `#include "PCH.h"`
3. Create the precompiled header with some macro defined to some value: `ccache cl /YcPCH.h -DFOO=0 -c PCH.cpp`
4. (Optionally, to see that things at this point, compile source.cpp using the precompiled header and the same macro definition: `ccache cl /YuPCH.h -DFOO=0 -c source.cpp`)
5. Recreate the precompiled header with a different value for the macro: `ccache cl /YcPCH.h -DFOO=1 -c PCH.cpp`
6. Compile source.cpp using the new precompiled header and the new value: `ccache cl /YuPCH.h -DFOO=1 -c source.cpp`

### Actual behavior ###
At step 5, despite the change in value the precompiled header compiled from step 3 has been retrieved from cache. The log confirms that "Succeeded getting cached result".
This causes step 6 to use an outdated precompiled header and is manifested by a warning.
```
source.cpp(1): warning C4005: 'FOO': macro redefinition
source.cpp(1): note: 'FOO' previously declared on the command line
source.cpp(1): warning C4651: '/DFOO=0' specified for precompiled header but not for current compile
```

### Expected behavior ###
The creation of the precompiled header at step 5 should result in a cache miss and the real compiler should create a updated precompiled header.


### Environment ###
ccache version master.4cb66a6a 
Windows 10
MSVC

--- a/src/ccache.cpp
+++ b/src/ccache.cpp
@@ -1719,7 +1719,8 @@ hash_argument(const Context& ctx,
   // going to have any effect at all. For precompiled headers this might not be
   // the case.
   if (!direct_mode && !ctx.args_info.output_is_precompiled_header
-      && !ctx.args_info.using_precompiled_header) {
+      && !ctx.args_info.using_precompiled_header
+      && !ctx.args_info.generating_pch) {
     if (compopt_affects_cpp_output(args[i])) {
       if (compopt_takes_arg(args[i])) {
         i++;

